dbms/src/Functions/FunctionsStringReplace.h

```diff
@@ -280,7 +232,28 @@ class FunctionStringReplace : public IFunction
             const auto * col_replacement_const = typeid_cast<const ColumnConst *>(column_replacement.get());
             auto replacement = col_replacement_const->getValue<String>();
 
-            if (const auto * col = checkAndGetColumn<ColumnString>(column_src.get()))
+            bool col_const = column_src->isColumnConst();
```

Comment:
Use Const Generics to avoid these logics (and costs) are involved in other branches (i.e. non-const).

Reply:
i changed it to: 
```cpp
if (const auto * col_const = checkAndGetColumnConst<ColumnString>(column_src.get()))
```